cmd/iam-store.go

```diff
@@ -777,8 +777,8 @@ func (store *IAMStoreSys) GetGroupDescription(group string) (gd madmin.GroupDesc
 // ListGroups - lists groups. Since this is not going to be a frequent
 // operation, we fetch this info from storage, and refresh the cache as well.
 func (store *IAMStoreSys) ListGroups(ctx context.Context) (res []string, err error) {
-	cache := store.lock()
-	defer store.unlock()
+	cache := store.rlock()
+	defer store.runlock()
```

Comment:
This change is not right - it should be a write lock because we are updating cache from disk in this call as commented.